### ISSUE TITLE
Adding dotted notes, `=` lengths and tie support to `l` directives

### ISSUE THREAD

**Issue by exodustx0** [KungFuFurby/AddMusicKFF#1]
I was redirected here after hearing you were working on this, except didn't want to touch tie support? I would have added it long ago if I had any idea how to get AMK compiling, but I can do you a pull request to add this in, already got the code. 

Oh, and side-note, plus a reason why AMK _really_ does need replacing (working on it): you seem to have the AMK 1.0.8 before it was moderated, meaning you haven't imported the actual released 1.0.8 (it was further changed before it passed moderation).

**Comment by KungFuFurby** [comment]
You're right! A few files were further modified afterwards. I'll get that pushed in.

No-go on ties due to a conflict with the actual tie command, which is a valid N-SPC command. I have the = length part implemented on my end, and I haven't gotten to the dotted notes yet. I can compare and contrast on the pull request with my own copy if you want.

**Comment by KungFuFurby** [comment]
Although I merged the changes, I have not closed the issue yet, pending a confirmation to let the tie portion remain cut.

**Comment by KungFuFurby** [comment]
I'm adding sound effects to the scope of this issue, because I both never implemented them *and* there is way more to do there in addition to what this will do.

**Comment by KungFuFurby** [comment]
This is officially problematic because of the way substitution works. Namely, substitution only works at the start of each character prior to processing anything else and does not account for any characters in between. This results in a quirk where a rest followed by a define that is the same first character as a rest gets lost in the mix because it is combined. This quirk can work in reverse by failing to detect a character as the first character of a macro substitution. Unfortunately, adding the `=` and `.` characters to parsing these two count: the latter one having actually been done in an example case and thus must be kept to avoid song breakage.

This kind of problem is going to result in two things:
- I need to check any parser modifications between AMK versions prior to my fork. The last version that Kipernal developed to my knowledge was 1.0.2, and I couldn't find any significant changes on the C++-side between 1.0.5 and 1.0.8. I don't have a copy of 1.0, and it is not extant on SMWCentral.
- I either have to bump the parser version up to 4, or add a `doReplacement` call prior to checking the new characters in order to avoid this particular problem. There is no filtering of characters when doing macro replacements other than the quotation characters.
